integration-tests/endToEndFlow.spec.ts

```diff
@@ -0,0 +1,151 @@
+    expect(url).toBeDefined()
+    console.log('S3 URL: ' + url)
+    return url as string
+  }
+
+  const downloadResultsCSVFromLink = async (
+    s3Link: string
+  ): Promise<string> => {
+    try {
+      const response = await axios({ url: s3Link, method: 'GET' })
+      return response.data
+    } catch (error) {
+      throw 'Error downloading results csv from S3 link'
+    }
+  }
+
+  const waitForDownloadHash = async (zendeskId: string): Promise<string> => {
+    let downloadHash = await retrieveSecureDownloadDbRecord(zendeskId)
+    const maxAttempts = 30
+    let attempts = 0
+    while (!downloadHash && attempts < maxAttempts) {
+      attempts++
+      await pause(2000)
+      downloadHash = await retrieveSecureDownloadDbRecord(zendeskId)
+    }
+
+    if (attempts == maxAttempts) {
+      throw Error(
+        'Download hash not populated within reasonable time. Please check logs to ensure that data retrieval and query execution were successful'
+      )
+    }
+    expect(downloadHash).toBeDefined()
+    console.log(`DOWNLOAD HASH: ${downloadHash}`)
+    return downloadHash ? downloadHash : ''
+  }
+
+  beforeEach(async () => {
+    await deleteAuditDataWithPrefix(
+      AUDIT_BUCKET_NAME,
+      `firehose/${END_TO_END_TEST_DATE_PREFIX}`
+    )
+
+    await copyAuditDataFromTestDataBucket(
+      AUDIT_BUCKET_NAME,
+      `firehose/${END_TO_END_TEST_DATE_PREFIX}/01/${END_TO_END_TEST_FILE_NAME}`,
+      END_TO_END_TEST_FILE_NAME
+    )
+  })
+
+  it('Query matches data - CSV file containing query results can be downloaded', async () => {
+    const EXPECTED_ADDRESS_VALID_FROM_DATE = `"2014-01-01"`
+    const EXPECTED_BIRTH_DATE = `"1981-07-28"`
+    const EXPECTED_POSTALCODE = `"AB10 6QW"`
+
+    const zendeskId: string = await createZendeskTicket(
+      endToEndTestRequestDataWithMatch
+    )
+    await approveZendeskTicket(zendeskId)
+
+    const downloadHash = await waitForDownloadHash(zendeskId)
+
+    const secureDownloadPageHTML = await getSecureDownloadPageHTML(downloadHash)
+
+    console.log(secureDownloadPageHTML)
+    expect(secureDownloadPageHTML).toBeDefined()
+
+    const resultsFileS3Link = retrieveS3LinkFromHtml(secureDownloadPageHTML)
+    expect(resultsFileS3Link.startsWith('https')).toBeTrue
+
+    const csvData = await downloadResultsCSVFromLink(resultsFileS3Link)
+    console.log(csvData)
+
+    const rows = CSV.parse(csvData, { output: 'objects' })
+    console.log(rows)
+
+    expect(rows.length).toEqual(1)
+    expect(rows[0].event_id).toEqual(END_TO_END_TEST_EVENT_ID)
+    expect(rows[0].address_postalcode).toEqual(EXPECTED_POSTALCODE)
+    expect(rows[0].address_validfrom).toEqual(EXPECTED_ADDRESS_VALID_FROM_DATE)
+    expect(rows[0].birthdate_value).toEqual(EXPECTED_BIRTH_DATE)
+    expect(rows[0].name).toBeDefined()
+  })
+
+  it('Query does not match data - Empty CSV file should be downloaded', async () => {
+    const zendeskId: string = await createZendeskTicket(
```

Comment:
Given that we're going to change this in future so the user is informed that there was no data available for their query, I think maybe we shouldn't have this test, because we'll have to remove it later.

Reply:
Is that change coming as part of MVP? Should we leave this until the change? Happy to remove, but this could be handy to help spot any regressions in the meantime.

Reply:
Yeah, fair enough, just leave it then